src/data-workspace/data-details-sidebar/comment.test.js

```diff
@@ -28,19 +43,149 @@ describe('<Comment />', () => {
         ).not.toBeInTheDocument()
     })
 
-    it('renders the item comment once loaded', async () => {
+    it('renders the item comment', async () => {
+        const item = {
+            categoryOptionCombo: 'coc-id',
+            dataElement: 'de-id',
+            comment: 'This is a comment',
+        }
+
         const { getByRole, queryByRole, getByText } = render(
-            <CustomDataProvider>
-                <Comment comment="Some comment" />
-            </CustomDataProvider>
+            <Comment item={item} />
         )
 
-        expect(getByText('Some comment')).toBeInTheDocument()
+        expect(getByText('This is a comment')).toBeInTheDocument()
         expect(
             queryByRole('button', { name: 'Add comment' })
         ).not.toBeInTheDocument()
         expect(
             getByRole('button', { name: 'Edit comment' })
         ).toBeInTheDocument()
     })
+
+    it('shows a loading indicator when submitting a comment change', async () => {
+        useSetDataValueCommentMutation.mockImplementation(() => {
+            const [loading, setLoading] = useState(false)
+            return {
+                isLoading: loading,
+                mutate: () => setLoading(true),
+            }
+        })
+
+        const item = {
+            categoryOptionCombo: 'coc-id',
+            dataElement: 'de-id',
+            comment: 'This is a comment',
+        }
+
+        const { getByRole, queryByRole } = render(<Comment item={item} />)
+
+        userEvent.click(getByRole('button', { name: 'Edit comment' }))
+
+        await waitFor(() => {
+            expect(queryByRole('textbox')).toBeInTheDocument()
+        })
+
+        const input = getByRole('textbox')
+        fireEvent.change(input, {
+            target: { value: 'Changed' },
+        })
+        expect(input.value).toBe('Changed')
+
+        expect(queryByRole('progressbar')).not.toBeInTheDocument()
+        userEvent.click(getByRole('button', { name: 'Save comment' }))
+        await waitFor(() => {
+            expect(getByRole('progressbar')).toBeInTheDocument()
+        })
```

Comment:
nit (feel free to ignore): this could be `expect(findByRole('progressbar')).not.toBeNull()` without the need for await waitFor